pgpubsub/channel.py

```diff
@@ -213,6 +232,45 @@ def _build_model_serializer_data(cls, payload: Dict, state: str):
         return model_data
 
 
+TX_ABORTED_ERROR_MESSAGE = (
+    'current transaction is aborted, commands ignored until end of transaction block'
+)
+
+def set_notification_context(
+    context: Dict[str, Any], using: Optional[str] = None
+) -> None:
+    if using:
+        conn = connections[using]
+    else:
+        conn = connection
+    if conn.needs_rollback:
+        return
+    use_tx_bound_notification_context = getattr(
+        settings, 'PGPUBSUB_TX_BOUND_NOTIFICATION_CONTEXT', False
+    )
+    if use_tx_bound_notification_context and not conn.in_atomic_block:
+        raise RuntimeError(
+            'Transaction bound context can be only set in atomic block. '
+            'Either start transaction with `atomic` or do not use transaction bound '
+            'payload context via PGPUBSUB_TX_BOUND_NOTIFICATION_CONTEXT=False'
+        )
+    with conn.cursor() as cursor:
+        try:
+            if use_tx_bound_notification_context:
+                scope = 'LOCAL'
+            else:
+                scope = 'SESSION'
+            cursor.execute(
+                f'SET {scope} pgpubsub.notification_context = %s',
```

Comment:
I'm a little bit confused here so probably a stupid question. What is actually being set here - where does `pgpubsub.notification_context` live in the db?

Reply:
This is a [custom parameter](https://www.postgresql.org/docs/15/runtime-config-custom.html). They are like custom defined settings that are stored either for a connection (scope='SESSION') or for a transaction (scope='LOCAL').